examples/v2_dry_run.rs

```diff
@@ -0,0 +1,117 @@
+//! Example of dry-run functionality of the node.
+
+use anyhow::Context;
+use clap::AppSettings;
+use concordium_base::{
+    base::Energy,
+    common::types::Timestamp,
+    contracts_common::{Address, Amount, ContractAddress, EntrypointName},
+    smart_contracts::{OwnedParameter, OwnedReceiveName},
+    transactions::Payload,
+};
+use concordium_rust_sdk::{
+    types::smart_contracts::ContractContext,
+    v2::{self, dry_run::DryRunTransaction, BlockIdentifier},
+};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:20000"
+    )]
+    endpoint: v2::Endpoint,
+}
+
+/// Test all dry run operations.
+async fn test_all(endpoint: v2::Endpoint) -> anyhow::Result<()> {
+    // Connect to endpoint.
+    let mut client = v2::Client::new(endpoint).await.context("Cannot connect.")?;
+    // Start the dry run session.
+    let mut dry_run = client.dry_run().await?;
```

Comment:
Since `load_block_state` is required as the first message, we could change the function `dry_run` to take a block identifier and have it call it straight away, such that the API prevents the user from doing anything else first.

Reply:
I have changed `dry_run` to load the block, and `begin_dry_run` now has the old behaviour.